src/lib/expression/abstract_expression.hpp

```diff
@@ -47,6 +47,7 @@ class AbstractExpression : public std::enable_shared_from_this<AbstractExpressio
  public:
   explicit AbstractExpression(const ExpressionType init_type,
                               const std::vector<std::shared_ptr<AbstractExpression>>& init_arguments);
+  AbstractExpression(const AbstractExpression&) = default;
```

Comment:
In which case would we want to copy an expression? if we actually need a second version, we probably want to use `deep_copy`. Should expressions maybe even be `Noncopyable`?

Reply:
Haven't worked with Expressions a lot. Whatever makes sense.

Reply:
Could you paste the compiler error that comes without your change?

Reply:
That's the one in the description.